### PR TITLE
Enable the use of an assumed role when talking to Vault

### PR DESCRIPTION
**Pull request by lattwood**

### CODE REVIEW

**Review by jhorowitz — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -7,12 +7,12 @@ The only supported Trusted Third Party is currently AWS IAM.
```
Comment (rikonor) [inline README.md:7]: In the README, I would add a bit of motivation for using this tool. I feel like it's not entirely clear.

Comment (rikonor) [reply README.md:7]: Not a blocker, but more comments in the code would be great.

File: vault/aws/iam.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	creds := stscreds.NewCredentials(sess, roleArn)
+	return &generator{
+		sts:       sts.New(sess, &aws.Config{Credentials: creds}),
+		vaultAddr: vaultAddress,
+	}
+}
+
+// WrappedToken returns a token for the role provided with the wrapTTL provided
+// WARNING: If wrapTTL is a zero length duration we provide an unwrapped token
+func (a generator) WrappedToken(role string, wrapTTL time.Duration) (*string, error) {
```
Comment (rikonor) [inline vault/aws/iam.go]: I think usually it's preferred to return `"", err` (meaning the method returns `(string, error)`) unless there's a special reason to return a `*string`.

Thinking, as a potential user of a library, that it's not as nice having to work with pointers to primitive types.

Comment (rikonor) [reply vault/aws/iam.go]: Also, any reason not to define the method on a pointer receiver? `func (a *generator) ...`

Comment (lattwood) [reply vault/aws/iam.go]: Not modifying any of the struct members so no pointer receiver.

Comment (lattwood) [reply vault/aws/iam.go]: Point well taken around `*string`, that's awkward. Will fix.

File: README.md (hunk 1 above)
Comment (lattwood) [reply README.md:7]: README.md will be improved in another PR

**Review by rikonor — APPROVED** [review]

**Pull request merged**